firebase-firestore/src/main/java/com/google/firebase/firestore/local/MemoryRemoteDocumentCache.java

```diff
@@ -112,6 +113,11 @@ public MaybeDocument get(DocumentKey key) {
         continue;
       }
 
+      SnapshotVersion readTime = entry.getValue().second;
+      if (readTime.compareTo(sinceUpdateTime) <= 0) {
```

Comment:
If there exist documents with a read time of zero and you pass `SnapshotVersion.MIN`, won't this skip them?

Reply:
The migration in #686 didn't populate read time, so all existing documents in the cache will have a zero read time.

Could you add tests for how this is expected to work against pre-existing documents?

Reply:
We don't really have good testing infrastructure (yet) that tests client functionality after migration. We have some tests that pretend to do client like queries before and after a schema migration, but they are all low level. I'm not opposed to adding tests like this, but:
- the MemoryRemoteDocumentCache will not have any existing documents and is not affected by migrations
- the SQLiteRemoteDocumentCache reads all documents with NULL version numbers.

Let me know if you want me to add a test.

Reply:
At some level we should prove to ourselves that this works and that it's going to stay working. I see the NULL check on the read_time so I think you're right that this is functional.

On some level this could just be a test that validates the migration behaves as we expect. If it's straightforward to do so, it would be useful to stand up a `SQLiteRemoteDocumentCache` in the `SQLiteSchemaTest` and validate that migrated documents always match queries regardless of the update time you pass.

If you also deleted NoDocuments with a zero time in that migration you could validate that they don't exist post as well.

Reply:
Also, to be clear: If it costs too much time or effort to make it work I'm fine with skipping it but I think it's probably worth trying.

Reply:
I hacked together a test. It wasn't as bad as I thought it would be, but in return it is also not very pretty :)